src/generators/Generator.ts

```diff
@@ -40,6 +41,8 @@ export default class Generator {
 	cssId: string;
 	usesRefs: boolean;
 
+	selectors: any[]; // TODO how to indicate it takes `(Node[]) => boolean` functions?
```

Comment:
Good catch. Have refactored it a bit so that it's a bit more TS-friendly (and possibly a bit clearer anyway) — a `Selector` class with an `apply` method